src/appimagetool_sign.c

```diff
@@ -231,10 +233,10 @@ bool embed_data_in_elf_section(const char* filename, const char* elf_section, gp
     char data_buffer[data_size];
 
     size_t total_bytes_read = 0;
-    size_t bytes_read = 0;
+    ssize_t bytes_read = 0;
 
     for (;;) {
-        bytes_read = gpgme_data_read(data, data_buffer, data_size);
+        bytes_read = gpgme_data_read(data, data_buffer + total_bytes_read, (size_t)(data_size - (off_t)total_bytes_read));
```

Comment:
Interesting. This appears to have been a bug before, I think.

I have no idea what happens if `(size_t)(data_size - (off_t)total_bytes_read)` is 0 though.